src/special/gamma.jl

```diff
@@ -0,0 +1,63 @@
+# This file is heavlily adopted from https://github.com/JuliaMath/SpecialFunctions.jl.
+# License is MIT: http://julialang.org/license
+
+function lgamma(x)
+  return CUDAnative.lgamma(x)
+end
+
+function digamma(x)
+  if x <= 0 # reflection formula
+    ψ = -π / CUDAnative.tan(π * x)
+    x = 1 - x
+  else
+    ψ = zero(x)
+  end
+  if x < 7
+    # shift using recurrence formula
+    ν = one(x)
+    n = 7 - CUDAnative.floor(x)
+    while ν <= n - 1
+      ψ -= inv(x + ν)
+      ν += one(x)
+    end
+    ψ -= inv(x)
+    x += n
+  end
+  t = inv(x)
+  ψ += CUDAnative.log(x) - 0.5 * t
+  t *= t # 1/z^2
+  # the coefficients here are Float64(bernoulli[2:9] .// (2*(1:8)))
+  ψ -= t * @evalpoly(t,0.08333333333333333,-0.008333333333333333,0.003968253968253968,-0.004166666666666667,0.007575757575757576,-0.021092796092796094,0.08333333333333333,-0.4432598039215686)
```

Comment:
Btw, this will convert into `Float64`; maybe it ought to be dependent on the input or use `Float32` by default?